### DISCUSSION TITLE
Is it possible to serve dynamic sitemap on each new request?

### DISCUSSION THREAD

**Discussion by bakirgdev** [vercel/next.js#73814] [Help] [answered]
### Summary

I need my sitemap to fetch up-to-date data and use it on each request, not only on build-time!

I tried to utilize the Dynamic API with server components but got:
- in build console output sitemap is a static route (hollow circle, not lambda)
- when the API endpoint response changes nothing sitemap stays the same

I wanted to find out here if this was even available before creating any issues

Make sure to check the example code

### Additional information

EDIT: NextJS v 15.1.0 w/ clean setup

### Example

```js
import type { MetadataRoute } from 'next';

export default async function Sitemap(): Promise<MetadataRoute.Sitemap> {
	const users = await fetch(`${process.env.API_URL}/users/list`).then((res) => res.json());

	const usersRoutes = users.map((user: { slug: string }) => ({
		url: `${process.env.APP_URL}/user/${user.slug}`,
		lastModified: new Date(),
		changeFrequency: 'monthly',
		priority: 0.8,
	}));

	return [
		{
			url: `${process.env.APP_URL}`,
			lastModified: new Date(),
			changeFrequency: 'monthly',
			priority: 1,
		},
		...usersRoutes,
	];
}
```

**Comment by icyJoseph** [comment, marked as answer]
Shouldn't you be able to add `export const dynamic = 'force-dynamic'`?

**Reply by bakirgdev** [reply]
Yep, totally forgot that. That solved the problem, thanks Jo!
